packages/cli/src/commands/generate/outputs/generate_outputs_command.test.ts

```diff
@@ -249,3 +254,101 @@ void describe('generate outputs command', () => {
     assert.match(output, /Arguments .* mutually exclusive/);
   });
 });
+
+void describe('GenerateOutputsCommand error handling', () => {
+  let clientConfigGenerator: ClientConfigGeneratorAdapter;
+  let backendIdentifierResolver: AppBackendIdentifierResolver;
+  let generateOutputsCommand: GenerateOutputsCommand;
+
+  beforeEach(() => {
+    // Mock the dependencies
+    clientConfigGenerator = {
+      generateClientConfigToFile: mock.fn(),
+    } as unknown as ClientConfigGeneratorAdapter;
+
+    backendIdentifierResolver = {
+      resolveDeployedBackendIdentifier: mock.fn(),
+    } as unknown as AppBackendIdentifierResolver;
+
+    generateOutputsCommand = new GenerateOutputsCommand(
+      clientConfigGenerator,
+      backendIdentifierResolver
+    );
+  });
+
+  void it('should throw AmplifyUserError when NO_APP_FOUND_ERROR occurs', async () => {
+    // Mock the resolver to simulate successful resolution
+    mock.method(
+      backendIdentifierResolver,
+      'resolveDeployedBackendIdentifier',
+      () => Promise.resolve({ appId: 'test-app', branchName: 'main' })
+    );
+
+    // Mock the generator to throw NO_APP_FOUND_ERROR
+    mock.method(clientConfigGenerator, 'generateClientConfigToFile', () => {
+      throw new BackendOutputClientError(
+        BackendOutputClientErrorType.NO_APP_FOUND_ERROR,
+        'No Amplify app found in the specified region'
+      );
+    });
+    try {
+      await generateOutputsCommand.handler({
+        stack: undefined,
+        appId: 'test-app',
+        'app-id': 'test-app',
+        branch: 'main',
+        format: undefined,
+        outDir: undefined,
+        'out-dir': undefined,
+        outputsVersion: '1.3',
+        'outputs-version': '1.3',
+        _: [],
+        $0: 'command-name',
+      });
+      assert.fail('Expected error was not thrown');
```

Comment:
use `commandRunner.runCommand` as used in other tests.

Reply:
updated it